### DISCUSSION TITLE
"Device does not support DeviceOrientationEvent"

### DISCUSSION THREAD

**Discussion by Icyyfr** [vsharma-git/virtualHOTAS#2] [Q&A] [answered]
Browser gives me that pop-up message after connecting to the address. The sliders and buttons work, but the gyro controls don't.

**Comment by vsharma-git** [comment]
It's great that at least the buttons are working. 

If you're getting the gyro message, that means your device doesn't have a gyro or the device isn't allowing the browser to access the gyro. 

You can go to this link to see if your device's gyro can be accessed by your browser. https://progressier.com/pwa-capabilities/device-orientation-event

**Reply by Icyyfr** [reply, marked as answer]
Well I tried virtualHOTAS on my other phone which had no issues whatsoever. I checked out the link you sent on the previous one and it did read my gyro inputs in the browser. I'll probably not look into it further though because I already got it working on this phone.

**Reply by vsharma-git** [reply]
Great! If you have any other questions, feedback or ideas please feel free to reach out.

**Comment by maximxlss** [comment]
I actually found the culprit, probably. See this: https://stackoverflow.com/questions/57663033/how-do-you-access-the-motion-sensors-accelerometer-gyroscope-in-a-mobile-brow
For me the ip was in 10.0.0.0 subnet (over vpn) and chrome didn't allow motion sensors, maybe it allows them for 192.0.0.0
